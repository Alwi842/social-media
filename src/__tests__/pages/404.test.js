import { render, screen, fireEvent } from "@testing-library/react";
import NotFound from "@/pages/404";
import { useRouter } from "next/router";
import "@testing-library/jest-dom";

// Mock Next.js router
jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

describe("404 Page", () => {
  let mockBack, mockPush;

  beforeEach(() => {
    // Mock router functions
    mockBack = jest.fn();
    mockPush = jest.fn();
    useRouter.mockReturnValue({
      back: mockBack,
      push: mockPush,
    });
  });

  it("renders correctly and matches snapshot", () => {
    const { container } = render(<NotFound />);

    // Check if the 404 text appears
    expect(screen.getByText("4 0 4")).toBeInTheDocument();
    expect(screen.getByText("Sorry, we couldn't find what you are looking for!")).toBeInTheDocument();

    // Snapshot test
    expect(container).toMatchSnapshot();
  });

  it("calls router.back() when 'Go Back' button is clicked", () => {
    render(<NotFound />);

    const goBackButton = screen.getByText("Go Back");
    fireEvent.click(goBackButton);

    expect(mockBack).toHaveBeenCalled();
  });

  it("calls router.push('/') when 'Home' button is clicked", () => {
    render(<NotFound />);

    const homeButton = screen.getByText(/Home/i);
    fireEvent.click(homeButton);

    expect(mockPush).toHaveBeenCalledWith("/");
  });
});
