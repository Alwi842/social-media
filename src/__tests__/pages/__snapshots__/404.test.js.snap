// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page renders correctly and matches snapshot 1`] = `
<div>
  <style>
    
          .emoji-404 {
            position: relative;
            animation: mymove 2.5s infinite;
          }
          @keyframes mymove {
            33% { top: 0px; }
            66% { top: 20px; }
            100% { top: 0px; }
          }
        
  </style>
  <div
    class="flex flex-col items-center justify-center min-h-screen bg-gray-100"
  >
    <svg
      class="emoji-404"
      enable-background="new 0 0 226 249.135"
      height="249.135"
      id="Layer_1"
      overflow="visible"
      viewBox="0 0 226 249.135"
      width="226"
      xml:space="preserve"
    >
      <circle
        cx="113"
        cy="113"
        fill="#FFE585"
        r="109"
      />
      <circle
        cx="68.732"
        cy="93"
        fill="#6E6E96"
        r="9"
      />
      <circle
        cx="156.398"
        cy="93"
        fill="#6E6E96"
        r="9"
      />
      <ellipse
        cx="67.732"
        cy="140.894"
        fill="#FF0000"
        opacity="0.18"
        rx="17.372"
        ry="8.106"
      />
      <ellipse
        cx="154.88"
        cy="140.894"
        fill="#FF0000"
        opacity="0.18"
        rx="17.371"
        ry="8.106"
      />
    </svg>
    <div
      class="text-center mt-6"
    >
      <h1
        class="text-gray-500 text-6xl font-bold tracking-widest"
      >
        4 0 4
      </h1>
      <p
        class="text-gray-500 text-xl mt-2"
      >
        Sorry, we couldn't find what you are looking for!
      </p>
    </div>
    <button
      class="mt-6 px-6 py-3 text-lg font-mono text-gray-500 bg-gray-200 rounded-md hover:shadow-md transition-all"
    >
      Go Back
    </button>
    <button
      class="mt-6 px-6 py-3 text-lg font-mono text-gray-500 bg-gray-200 rounded-md hover:shadow-md transition-all"
    >
      <svg
        fill="#e8eaed"
        height="24px"
        viewBox="0 -960 960 960"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M240-200h120v-240h240v240h120v-360L480-740 240-560v360Zm-80 80v-480l320-240 320 240v480H520v-240h-80v240H160Zm320-350Z"
        />
      </svg>
       Home
    </button>
  </div>
</div>
`;
